components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOFacade.java

```diff
@@ -0,0 +1,331 @@
+     * @param tenantId  Tenant ID.
+     * @return Webhook subscription.
+     * @throws WebhookMgtException If an error occurs while retrieving the webhook subscription.
+     */
+    @Override
+    public Webhook getWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+
+        return webhookManagementDAO.getWebhook(webhookId, tenantId);
+    }
+
+    /**
+     * Update a webhook subscription in the database.
+     * First updates the webhook subscription with webhook subscriber service,
+     * then updates its persistent state.
+     *
+     * @param webhookId Webhook subscription ID.
+     * @param webhook   Updated webhook subscription.
+     * @param tenantId  Tenant ID.
+     * @throws WebhookMgtException If an error occurs while updating the webhook subscription.
+     */
+    @Override
+    public void updateWebhook(String webhookId, Webhook webhook, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        EventSubscriberService subscriberService =
+                WebhookManagementComponentServiceHolder.getInstance().getEventSubscriberService();
+
+        try {
+            jdbcTemplate.withTransaction(template -> {
+
+                // Get the existing webhook
+                Webhook existingWebhook = webhookManagementDAO.getWebhook(webhookId, tenantId);
+
+                if (existingWebhook == null) {
+                    throw WebhookManagementExceptionHandler.handleClientException(
+                            ErrorMessage.ERROR_CODE_WEBHOOK_NOT_FOUND);
+                }
+
+                // If endpoint is being changed, check if the new endpoint already exists
+                if (!existingWebhook.getEndpoint().equals(webhook.getEndpoint()) &&
+                        isWebhookEndpointExists(webhook.getEndpoint(), tenantId)) {
+                    throw WebhookManagementExceptionHandler.handleClientException(
+                            ErrorMessage.ERROR_CODE_WEBHOOK_ALREADY_EXISTS);
+                }
+
+                // First unsubscribe the existing webhook
+                try {
+                    subscriberService.unsubscribe(existingWebhook, tenantDomain);
+                } catch (WebhookMgtException e) {
+                    LOG.warn("Error unsubscribing webhook during update: " + existingWebhook.getId(), e);
+                }
+
+                // Then try to subscribe with the updated webhook
+                webhook.setId(webhookId);
+                try {
+                    if (WebhookStatus.ACTIVE.equals(webhook.getStatus())) {
+                        boolean subscribed = subscriberService.subscribe(webhook, tenantDomain);
+                        if (!subscribed) {
+                            webhook.setStatus("INACTIVE");
+                        }
+                    }
```

Comment:
Have the same concern I added with regard to creation. And it's not a good practice to set active or inactive based on how system works. It should be some thing the user decides to change